bin/genesis

```diff
@@ -22,6 +22,10 @@ our $VERSION = "(development)";
 our $GITHUB_URL = "https://github.com/starkandwayne/genesis";
 our $USER_AGENT_STRING = "genesis/$VERSION";
 
+if ($ENV{BOSH_ALL_PROXY} ne '') {
```

Comment:
If BOSH_ALL_PROXY isn't set, this throws an uninitialized value, since you are comparing `undef` to the empty string.  You can simplify by just using `$ENV{BOSH_ALL_PROXY}` as your conditional -- in Perl, the empty string, false, 0, and the undefined value all evaluate to false in truthy context.

Reply:
Lovely.